low-level/files/opt/seagate/sspl/setup/resource_map/lr2/server.py

```diff
@@ -463,3 +480,115 @@ def get_nw_status(self, nw_interface, interface):
             nw_cable_conn_status = "UNKNOWN"
             logger.exception(self.log.svc_log(err))
         return nw_status, nw_cable_conn_status
+
+    def get_cortx_service_info(self):
+        """Get cortx service info in required format."""
+        service_info = []
+        cortx_services = Service().get_cortx_service_list()
+        for service in cortx_services:
+            response = self.get_systemd_service_info(service)
+            if response is not None:
+                service_info.append(response)
+        return service_info
+
+    def get_external_service_info(self):
+        """Get external service info in required format."""
+        service_info = []
+        external_services = Service().get_external_service_list()
+        for service in external_services:
+            response = self.get_systemd_service_info(service)
+            if response is not None:
+                service_info.append(response)
+        return service_info
+
+    def get_systemd_service_info(self, service_name):
+        """Get info of specified service using dbus API."""
+        try:
+            unit = Service()._bus.get_object(
+                SYSTEMD_BUS, Service()._manager.LoadUnit(service_name))
+            properties_iface = Interface(unit, dbus_interface=PROPERTIES_IFACE)
+        except DBusException as err:
+            logger.error(self.log.svc_log(
+                f"Unable to initialize {service_name} due to {err}"))
+            return None
```

Comment:
can simply use return statement.
why return None?

Reply:
We can, but it's a standard practice to return None if the function can return other types as per given conditions.

Reply:
Are we expecting any other return values for future use-case?
if no, then only keep return statement

Reply:
You didn't get my point. This function will also return a dictionary if everything goes right.
See this discussion: https://stackoverflow.com/a/15300582/7544320

_I think that it is reasonably idiomatic to leave off the return statement unless you need it to break out of the function early (in which case a bare return is more common), or return something other than None. It also makes sense and seems to be idiomatic to write return None when it is in a function that has another path that returns something other than None. Writing return None out explicitly is a visual cue to the reader that there's another branch which returns something more interesting (and that calling code will probably need to handle both types of return values)._